pairs/src/Pair.ts

```diff
@@ -272,13 +275,37 @@ export function handleSwap(event: SwapEvent): void {
       if (usdprice != BigDecimalZero) {
         swap.close = usdprice
       }
+
       swap.volume = swap.volume.plus(volume.times(usdprice))
       swap.lpfees = swap.lpfees.plus(lpfees.times(usdprice))
       swap.klimaearnedfees = swap.klimaearnedfees.plus(klimaearnedfees.times(usdprice))
       swap.slippage = swap.slippage.plus(slippage.times(usdprice))
       swap.save()
     }
-    pair.currentprice = swap.close
+
+    // calculate cco2 fee from contract. Apply to currentprice and currentPricePerTonne
+    if (event.address == KLIMA_CCO2_PAIR) {
+      let cco2_contract = CCO2TokenContract.bind(CCO2_ERC20_CONTRACT)
+      let decimalRatio = BigDecimal.fromString(cco2_contract.decimalRatio().toString())
+      let transactionPercentage = BigDecimal.fromString(cco2_contract.transactionPercentage().toString())
```

Comment:
This should be burning percentage. 

Currently we are not factoring in for the transactionPercentage.. as Coorest has decided to keep txnFee => 0. 

Note: transactionPercentage != burningPercentage..

Reply:
Unless txnPercentage is used for any other purpose.